internal/dag/secret.go

```diff
@@ -50,35 +51,73 @@ func isValidSecret(secret *v1.Secret) (bool, error) {
 }
 
 func validateCertificate(data []byte) error {
-	cert, err := decodePEM(data)
-	if err != nil {
-		return err
+	var exists bool
+	var block *pem.Block
+	for len(data) > 0 {
+		block, data = pem.Decode(data)
+		if block == nil {
+			return errors.New("failed to parse PEM block")
+		}
+		if !strings.HasSuffix(block.Type, "CERTIFICATE") {
```

Comment:
I'm sorry this has dragged on but the discussion in `validatePrivateKey` makes me wonder if this logic is correct. This skips checking blocks that don't end in certificate, thereby passing them to envoy without validation.

I think that two things should happen

1.  Rather than strings.HasSuffix this code should mirror validatePrivateKey and use a switch on the expected block types. 
2. What other kinds of blocks are present in public certificates? ISTM that there shouldn't be any, and the underlying requirement is to support certificates with some intermediate CAs included in the pem file (apologies for the incorrect terminology, I'm a noob).